frontend/src/components/ChatApp/MessageList.tsx

```diff
@@ -0,0 +1,51 @@
+import { useState, useEffect, useRef } from 'react';
+import MessageItem from './MessageItem';
+import { db } from '../../../../backend/firebase/firebase'
+import { collection, getDocs, query, orderBy, DocumentData } from "firebase/firestore"
+
+const MessageList: React.FC = () => {
+  const messagesRef = collection(db, "Chats");
+  const queryDoc = query(messagesRef, orderBy('createdAt'));
+  const [messages, setMessages] = useState<DocumentData[]>([]);
+  // Ref to scroll to most recent message in MessageList
+  const scrollEnd = useRef<null | HTMLDivElement>(null);
+
+  useEffect(() => {
+    const getMessages = async () => {
+      const querySnapShot = await getDocs(queryDoc);
```

Comment:
could be good to handle fetching errors so users know whats going on if it happens instead of just having the app crash on them